sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -40,6 +41,13 @@ sealed abstract class Filter {
     case f: Filter => f.references
     case _ => Array.empty
   }
+
+  /**
+   * If any of the references of this filter contains nested column
+   */
+  private[sql] def containNestedColumn: Boolean = {
```

Comment:
Minor: other places (i.e. `Collection#contains`) normally use `contains` instead of `contain`.